### PR TITLE
Framework for performing parameter sweep optimization

### PR DESCRIPTION
**Pull request by ZanderKeith**
Using the Optuna library we can perform an efficient search of the model parameter space (including dimensions, amount of slurry in the burner tank, FLiBe enrichment, etc) using Bayesian inference. I've used this library in the past for hyperparameter tuning machine learning models, but it should also be applicable here.

This included a major rewrite to the simple_geometry.py file to make it a function instead of just a script. Still have not included actual slurry contents, however that should be relatively straightforward.

### CODE REVIEW

**Comment by RemDelaporteMathurin** [comment]
I see some commits from Joe here, maybe you need to rebase your branch

**Comment by ZanderKeith** [comment]
> I see some commits from Joe here, maybe you need to rebase your branch

What is the motivation for doing a rebase in this instance? I pulled changes not present in the primary repository from Joe's fork to get his version of model creation before modifying it, so some of his commits show up in the pull request. If I rebase to the primary repo which still doesn't have those changes would anything be different?

**Comment by RemDelaporteMathurin** [comment]
Oh ok I didn't understand that. So in that case, it would be preferable to merge #5 first since this PR brings the right changes to main, then, once it's merged (and maybe modified since there were some comments), rebase this PR on j-fletcher:main and merge.

Does that make sense?

File: run_optimization.py [hunk 1]
```diff
@@ -96,19 +96,26 @@ def main(sweep_config_path, num_trials=1):
         Number of trials to run. This will add num_trials to the existing trials in the sweep_results.db
     """
 
+    # Change to the sweep directory
```
Comment (eepeterson) [inline run_optimization.py:110]: Porbably want a context manager to handle switching directories for you. [See here for an example](https://stackoverflow.com/questions/431684/equivalent-of-shell-cd-command-to-change-the-working-directory/431747#431747). Or [this answer](https://stackoverflow.com/questions/41742317/how-can-i-change-directory-with-python-pathlib) on how to create your own context manager using a decorator. I'd try the first one first.

File: run_optimization.py [hunk 2]
```diff
@@ -122,5 +129,5 @@ def main(sweep_config_path, num_trials=1):
     elif len(sys.argv) == 3:
         main(sys.argv[1], int(sys.argv[2]))
     else:
-        print("Usage: python run_optimization.py <sweep_config_path> <num_trials>")
+        print("Usage: python run_optimization.py <relative_sweep_directory> <num_trials>")
         sys.exit(1)
```
Comment (eepeterson) [inline run_optimization.py]: Generally calling `sys.exit` is not good practice. If you're writing a script in Python and you want it to error out, you raise an `Exception` or `Error` relevant to the issue.

File: run_optimization.py [hunk 3]
```diff
@@ -122,5 +129,5 @@ def main(sweep_config_path, num_trials=1):
     elif len(sys.argv) == 3:
```
Comment (eepeterson) [inline run_optimization.py]: I know these are already existing issues, and not part of this PR, but we should move to the argparse module if we want to be running scripts from the command line. See how OpenMC does this [here](https://github.com/openmc-dev/openmc/blob/develop/scripts/openmc-track-to-vtk). Sorry I didn't catch all this in earlier PRs.

**Review by eepeterson — CHANGES_REQUESTED** [review]
Should probably start reorganizing a CLI around best practices

File: run_optimization.py (hunk 1 above)
Comment (ZanderKeith) [reply run_optimization.py:110]: That's cool, didn't know you could do that. I foresee using this functionality elsewhere so I will put it in its own module (maybe to be expanded in the future).
